update_scanners.py

```
@@ -470,18 +470,28 @@ def trivy_cache_dir() -> Path | None:
+def open_https(request: urllib.request.Request, timeout: int) -> Any:
+    """Open an HTTPS request, refusing any other scheme.
+
+    urllib follows file:// and ftp:// as readily as https://, so a URL that
+    reaches this program through configuration gets its scheme checked here
+    rather than trusted.
+    """
+    if not request.full_url.startswith("https://"):
+        raise ValueError(f"refusing non-HTTPS URL: {request.full_url}")
+    return urllib.request.urlopen(  # nosec B310  # nosemgrep
+        request, timeout=timeout
+    )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)